### PR TITLE
Drop PHP before 7.4

### PR DESCRIPTION
**Pull request by phil-davis**
1) Update tools and apply cs-fixer suggested changes
2) Add type declarations to code
3) Add type declarations to tests
4) Add changelog and major version bump

### CODE REVIEW

File: .php-cs-fixer.dist.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-$config = PhpCsFixer\Config::create();
+$config = new PhpCsFixer\Config();
 $config->getFinder()
     ->exclude('vendor')
     ->in(__DIR__);
```
Comment (staabm) [inline .php-cs-fixer.dist.php:6]:
```suggestion
    ->in(__FILE__)
    ->in(__DIR__);
```

**Review by staabm — APPROVED** [review]
one nit, otherwise lgtm

File: .php-cs-fixer.dist.php (hunk 1 above)
Comment (phil-davis) [reply .php-cs-fixer.dist.php:6]: If I do this, then I get:
`The "/home/phil/git/sabre-io/cache/.php-cs-fixer.dist.php" directory does not exist.`

`in()` takes a directory/directories.

Comment (phil-davis) [reply .php-cs-fixer.dist.php:6]: In sabre-io/xml we put:
```
    ->append([
        __FILE__,
    ]);
```

I guess that makes cs-fixer parse-and-fix-format of its own `.php-cs-fixer.dist.php` code.

**Pull request merged**